### PR TITLE
HPCC4J-636 DFSClient: Improve Opentelemetry tracing

### PR DESCRIPTION
**Pull request by jpmcmu**
- Improved span names
- Transitioned read request events to read spans
- Move events to spans for connect, version and close
- Added span batch support

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [X] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by jpmcmu** [comment]
Screenshot with missing request added in: 
Screenshot 2024-08-30 at 2 52 33 PM

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java [hunk 1]
```diff
@@ -344,6 +335,35 @@ public HpccRemoteFileReader(FileReadContext ctx, DataPartition dp, IRecordBuilde
...
+        long readSize = context.readSizeKB;
+        if (readSize < 0)
+        {
+            readSize = RowServiceInputStream.DEFAULT_MAX_READ_SIZE_KB;
+        }
+        readSize *= 1000;
+
+        Attributes attributes = Attributes.of(  AttributeKey.stringKey("server.0.address"), primaryIP,
+                                                AttributeKey.stringKey("server.1.address"), secondaryIP,
+                                                ServerAttributes.SERVER_PORT, Long.valueOf(dp.getPort()),
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java:360]: minor (perhaps) but we've noticed APM/grafana tends to group numeric attributes seperately from string based attributes, forcing us to report ports as strings to keep them group with hostnames, etc.

**Review by rpastrana — APPROVED** [review]
@jpmcmu looks pretty good. Left a minor comment about reporting numeric values as strings, simply for presentation purposes

**Comment by jpmcmu** [comment]
Updated screenshot with timestamp fix:
Screenshot 2024-09-04 at 11 08 21 AM

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCRemoteFileWriter.java [hunk 2]
```diff
@@ -156,7 +156,7 @@ public HPCCRemoteFileWriter(FileWriteContext ctx, DataPartition dp, IRecordAcces
 
         Attributes attributes = Attributes.of(  AttributeKey.stringKey("server.0.address"), primaryIP,
                                                 AttributeKey.stringKey("server.1.address"), secondaryIP,
-                                                ServerAttributes.SERVER_PORT, Long.valueOf(dp.getPort()));
+                                                AttributeKey.stringKey("server.port"), "" + dp.getPort());
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCRemoteFileWriter.java:159]: unless this approach performs better, we should use explicit functionality like String.valueOf()

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java [hunk 3]
```diff
@@ -865,7 +867,7 @@ private void startNewReadRequestSpan()
             readRequestCount++;
 
             currentReadRequestEvent = new ReadRequestEvent();
-            currentReadRequestEvent.requestTime = System.nanoTime();
+            currentReadRequestEvent.requestTime = System.currentTimeMillis();
```
Comment (rpastrana) [inline dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java:870]: do we know why nano time doesn't work

**Review by rpastrana — COMMENTED** [review]
@jpmcmu just a couple of comments

File: dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java (hunk 3 above)
Comment (jpmcmu) [reply dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java:870]: I believe it was getting truncated by something in the OTEL stack

**Comment by jpmcmu** [comment]
@rpastrana Changed string conversion and squashed

**Pull request merged**